packages/flame/lib/src/effects2/transform2d_effect.dart

```diff
@@ -14,41 +12,14 @@ import 'effect_controller.dart';
 /// Currently this class only supports being attached to [PositionComponent]s,
 /// but in the future it will be extended to work with any [Transform2D]-based
 /// classes.
-abstract class Transform2DEffect extends Effect {
+abstract class Transform2DEffect extends ComponentEffect<PositionComponent> {
```

Comment:
I don't think this is correct: a `Transform2DEffect` targets a `Transform2D` property, which doesn't have to be on a `PositionComponent`. For example, I want the same effects to be able to target a `Camera` (once we refactor it to use the `Transform2D` internally). There could be other use cases too down the line.

Reply:
Right now the `Transform2DEffect` is using the `PositionComponent` to reach to the `Transform2D` though, and later we can refactor it to somehow differently reach the `Transform2D`. The most common use case will be through a component though, and later maybe the `Camera` will be a component that has a `Transform2D` too even.

Reply:
I mean, currently (in the `main` branch), `Transform2DEffect` is a class that already does its job, and also can be trivially extended to support `Camera` and other use cases. 

With your suggested changes, it can no longer be extended to support `Camera` and other use cases without doing a substantial refactoring, which would probably involve reverting to the state that the class is currently in.

At a fundamental level, `Transform2DEffect` is not a `ComponentEffect` because it does not target a component -- it targets a specific property, and it just so happens that `PositionComponent` has that property.

Reply:
As soon as you want to support `Transform2D` without a component, you just stop extending `ComponentEffect`. But right now it is tightly coupled with `PositionComponent` anyways, so why have code duplication?

Reply:
@st-pasha does it make sense? On main `Transform2DEffect` has this right now:
```dart
    if (parent is PositionComponent) {
      target = (parent! as PositionComponent).transform;
    }
    // TODO(st-pasha): add Camera support once it uses Transform2D
    else {
      throw UnsupportedError(
        'Can only apply a Transform2DEffect to a PositionComponent class',
      );
    }
```